pkgs/tools/networking/pihole/0002-Patch-and-remove-unsupported-commands.patch

```diff
@@ -0,0 +1,290 @@
+From 7264446f3c15712323c2915c31a74fe97913781d Mon Sep 17 00:00:00 2001
+From: williamvds <william@williamvds.me>
+Date: Sat, 1 Apr 2023 19:17:19 +0100
+Subject: [PATCH 2/3] Patch and remove unsupported commands
+
+Don't rely on sourcing the install script. Copy the required functions
+(valid_ip and valid_ip6) to the script that needs it.
+
+Remove some unsupported maintenance commands, particularly the ones which
+reinstall, update, and uninstall pihole. This is managed by NixOS, after all.
```

Comment:
We should probably just exit those commands early so that people don't wonder where they went.